compiler/src/iree/compiler/Dialect/Util/TransformOps/test/symbol_transforms.mlir

```diff
@@ -0,0 +1,23 @@
+// RUN: iree-opt --transform-interpreter %s --split-input-file --verify-diagnostics | FileCheck %s
+
+module attributes {transform.with_named_sequence} {
+  transform.named_sequence @__transform_main(%arg1: !transform.any_op {transform.readonly}) {
+    %0 = transform.util.get_sibling_symbol @__transform_main : !transform.any_op
```

Comment:
Why is this called `get_sibling_symbol` instead of, say, `lookup_op_with_symbol`? Can you give some example of when it won't return the immediately matching any_op?

The documentation says:
https://github.com/llvm/llvm-project/blob/aa3149d204ed019c53eedd1bb00a315c0db41270/mlir/include/mlir/IR/SymbolTable.h#L166-L169

Which to me seems like the nearest parent, not a sibling.

Reply:
I wanted to somehow specify that it is looking up in the symbol table of the transform op (not the payload IR), but sibling was probably the wrong word. Maybe `lookup_nearby_symbol`

Reply:
or follow the documentation and `lookup_nearest_symbol_from_self`

Reply:
The last one is verbose but by far the least confusing to me

Reply:
Seems like here too using nested symbols can resolve things automatically for you?

Reply:
LMK if the new name makes more sense.